protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -800,7 +807,12 @@ func (rpccs *RPCConsumerServer) relayInner(ctx context.Context, singleConsumerSe
 		metadataAdd := metadata.New(map[string]string{common.IP_FORWARDING_HEADER_NAME: consumerToken})
 		connectCtx = metadata.NewOutgoingContext(connectCtx, metadataAdd)
 		defer connectCtxCancel()
+
+		// add consumer processing timestamp before provider metric and start measuring time after the provider replied
+		go rpccs.rpcConsumerLogs.AddMetricForProcessingLatencyBeforeProvider(analytics, rpccs.listenEndpoint.ChainID, rpccs.listenEndpoint.ApiInterface)
```

Comment:
this one you dont want in a routine, as you add time.Since() there meaning it wont be accurate if it happens in a routine.

either use the time.Since before calling this method Or call a routine inside like I suggested in my other comment

Reply:
done. see https://github.com/lavanet/lava/commit/86b7f3173a1068378b2a1e327e9082bb37ef2a90